src/components/menu/menu.svelte

```diff
@@ -102,7 +103,9 @@
       })
     }
 
-    if (item.tagName === 'LEO-MENU-ITEM') {
+    // When using keyboard navigation to call selectMenuItem, ensure click handlers happen on the item,
+    // but not when this was called in a click handler, otherwise we'll get 2x clicks.
+    if (item.tagName === 'LEO-MENU-ITEM' && e.type !== 'click') {
```

Comment:
I'm a little lost here, but do we not call `click` on `leo-option`s too, when selecting them with the keyboard?

Reply:
No. I added a click call to leo-menu-item because we expect those to have their own click handler in the consumer, but for dropdown menus, we don't expect them to have any onClick handlers in the calling code - only Nala Dropdown code would perform the "select" action